### PR TITLE
spring-08 done

### PR DESCRIPTION
**Pull request by less08**

### CODE REVIEW

File: src/main/java/com/dev/cinema/config/SecurityConfig.java [hunk 1]
```diff
@@ -1,38 +1,44 @@
...
-            .anyRequest().authenticated()
-            .and()
-            .formLogin().permitAll()
-            .and()
-            .httpBasic()
-            .and()
-            .csrf().disable();;
+        http.authorizeRequests()
+                .antMatchers(HttpMethod.POST, "/register").permitAll()
+                .antMatchers(HttpMethod.POST).hasRole("ADMIN")
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/config/SecurityConfig.java]: check your `addSession()` and `completeOrder()`, should it be restricted only to ADMIN?

File: src/main/java/com/dev/cinema/dao/RoleDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Role getRoleByName(String roleName) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Role> query = session.createQuery("from Role where roleType=:roleName",
+                    Role.class).setParameter("roleName", roleName);
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/dao/RoleDaoImpl.java]:
```suggestion
                    Role.class).setParameter("roleName", RoleType.valueOf(roleName));
```
it won't work in other way

File: src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -1,28 +1,39 @@
...
         user.setPassword(password);
+        user.setRoles(getUserRoles());
         User userFromDB = userService.add(user);
         shoppingCartService.registerNewShoppingCart(userFromDB);
         return userFromDB;
     }
+
+    private List<Role> getUserRoles() {
+        return List.of(roleService.getRoleByName(RoleType.USER.toString()));
+    }
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java]: Why is this method in AuthenticationService? Also, is ADMIN prohibited to be an USER?

File: src/main/java/com/dev/cinema/service/UserDetailsServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        Optional<User> userOptional = userService.findByEmail(email);
+
+        if (userOptional.isEmpty()) {
+            throw new UsernameNotFoundException("User with email " + email + "was not found");
+        }
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/service/UserDetailsServiceImpl.java]: If you want to check it, use `orElseThrow` instead

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/config/SecurityConfig.java [hunk 5]
```diff
@@ -27,18 +27,29 @@ public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception
...
+            .antMatchers(HttpMethod.GET, "/movie-sessions/available").hasAnyRole("ADMIN","USER")
+            .antMatchers(HttpMethod.GET, "/movie-sessions/{id}").hasAnyRole("ADMIN","USER")
+            .antMatchers(HttpMethod.POST, "/movie-sessions").hasRole("ADMIN")
+            .antMatchers(HttpMethod.PUT, "/movie-sessions/{id}").hasRole("ADMIN")
+            .antMatchers(HttpMethod.DELETE, "/movie-sessions/{id}").hasRole("ADMIN")
+            .antMatchers(HttpMethod.GET, "/orders").hasRole("USER")
+            .antMatchers(HttpMethod.POST, "/orders/complete").hasRole("USER")
+            .antMatchers(HttpMethod.POST, "/shopping-carts/movie-sessions").hasRole("USER")
+            .antMatchers(HttpMethod.GET, "/shopping-carts/by-user").hasRole("USER")
+            .antMatchers(HttpMethod.GET, "/users/by-email").hasRole("ADMIN")
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/config/SecurityConfig.java:44]: 🥇

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/com/dev/cinema/model/User.java [hunk 6]
```diff
@@ -16,6 +18,8 @@
     @Column(unique = true)
     private String email;
     private String password;
+    @ManyToMany
+    private List<Role> roles;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/model/User.java:22]: it's better to use set here

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**